api/src/main/java/jakarta/data/repository/Pagination.java

```diff
@@ -118,24 +151,73 @@ public List<Sort> sorts() {
 
     @Override
     public Pageable newPage(long pageNumber) {
-        return new Pagination(pageNumber, size, sorts);
+        return new Pagination(pageNumber, size, sorts, mode, cursor);
     }
 
     @Override
     public Pageable newSize(int maxPageSize) {
-        return new Pagination(page, maxPageSize, sorts);
+        return new Pagination(page, maxPageSize, sorts, mode, cursor);
     }
 
     @Override
     public Pageable sortBy(Iterable<Sort> sorts) {
         List<Sort> sortList = sorts == null
                 ? Collections.emptyList()
                 : StreamSupport.stream(sorts.spliterator(), false).collect(Collectors.toUnmodifiableList());
-        return new Pagination(page, size, sortList);
+        return new Pagination(page, size, sortList, mode, cursor);
     }
 
     @Override
     public Pageable sortBy(Sort... sorts) {
-        return new Pagination(page, size, sorts == null ? Collections.emptyList() : List.of(sorts));
+        return new Pagination(page, size, sorts == null ? Collections.emptyList() : List.of(sorts), mode, cursor);
+    }
+
+    /**
+     * Built-in implementation of Cursor.
+     */
+    static final class CursorImpl implements Cursor {
```

Comment:
The same question to the inner-class.
At this point, we should also have a unit-class for it.

Reply:
I added commit https://github.com/jakartaee/data/pull/73/commits/dc7a6e61ba422de3253ef68cab2cd93eb02297fb to update it as you suggested.  Please review.